resources/Shaders/ExplosionEffect.geom.glsl

```diff
@@ -37,161 +42,128 @@ out VertexData{
+  return abs(sign(x - y));
+}
+
+vec3 GreaterThan(vec3 x, vec3 y) {
+  return max(sign(x - y), 0.0);
+}
+
+vec3 LessThan(vec3 x, vec3 y) {
+  return max(sign(y - x), 0.0);
+}
+
+vec3 GreaterEqualTo(vec3 x, vec3 y) {
+  return 1.0 - LessThan(x, y);
+}
+
+vec3 LessEqualTo(vec3 x, vec3 y) {
+  return 1.0 - GreaterThan(x, y);
+}
+
 // returns a "random" number based on input parameter
 float GetRandomNumber(int polygon_index)
 {
-	int randomIndex = int(mod(polygon_index, 50));
-	
-	return RandomNumbers[randomIndex];
+	return RandomNumbers[int(mod(polygon_index, 50))];
 }
 
-float randomNumber = 0.0;
-float randomDistance = 0.0;
-
-void main()
+vec3 CalcCenterOfTriangle(vec3 vertex0, vec3 vertex1, vec3 vertex2)
 {
 	// calculate middle of vectors
-	vec3 v1 = Input[1].Position - Input[0].Position;
-	vec3 v2 = Input[2].Position - Input[0].Position;
-	vec3 v3 = Input[2].Position - (v1 / 2);
-	vec3 v4 = Input[1].Position - (v2 / 2);
+	vec3 dir1 = normalize(vertex1 - vertex0);
+	vec3 dir2 = normalize(vertex2 - vertex0);
 	
-	// calculate intersection
-	
-	// normalize direction
-	vec3 dir1 = normalize(v1);
-	vec3 dir2 = normalize(v2);
-	
-	vec3 vecA = Input[2].Position - Input[1].Position; //o2 - o1
+	vec3 vecA = vertex2 - vertex1; //o2 - o1
 	vec3 vecB = cross(dir1, dir2);
 	
 	mat3 matrisA = mat3(vecA, dir2, vecB);
 	
 	float lengthA = length(vecB);
 	lengthA = lengthA * lengthA;
-
+	
 	float s = determinant(matrisA) / lengthA;
 	
 	// center position of triangle
-	vec3 centerOfTriangle = Input[1].Position + (s * dir1);
-	
+	return vertex1 + (s * dir1);
+}
+
+void PassThingsThrough(int index)
+{
+	// pass through vertex data
+	Output.Normal = Input[index].Normal;
+	Output.Position = Input[index].Position;
+	Output.TextureCoordinate = Input[index].TextureCoordinate;
+	Output.Tangent = Input[index].Tangent;
+	Output.BiTangent = Input[index].BiTangent;
+	Output.ExplosionColor = EndColor;	
+}
+
+void main()
+{
+	vec3 centerOfTriangle = CalcCenterOfTriangle(Input[0].Position, Input[1].Position, Input[2].Position);
+		
 	// center position of triangle to origin vector
 	vec3 origin2TriangleCenterVector = centerOfTriangle - ExplosionOrigin;
 	vec3 normalizedOrigin2TriangleCenterVector = normalize(origin2TriangleCenterVector);
 	
+	// distance between origin and the center of the current triangle
+	float origin2TriangleCenterDistance = length(origin2TriangleCenterVector);	
+	
 	// time percentage until end of explosion (0.0-1.0)
 	float timePercetage = TimeSinceDeath / ExplosionDuration;
 	
-		// get a random number if randomness is enabled, otherwise the random number will be zero and won't affect the other algorithms
-	if (Randomness == true)
-	{
-		randomDistance = GetRandomNumber(gl_PrimitiveIDIn) * RandomnessScalar;
-	}
+	// get a random number if randomness is enabled, otherwise the random distance will be zero and won't affect the other algorithms
+	float randomDistance = GetRandomNumber(gl_PrimitiveIDIn) * RandomnessScalar * float(Randomness);
```

Comment:
Nice one. Hopefully it actually works that way. In any case it doesn't break anything.